app/src/main/java/org/mozilla/tv/firefox/FirefoxApplication.kt

```diff
@@ -89,18 +90,14 @@ open class FirefoxApplication : LocaleAwareApplication() {
     }
 
     private fun initPush() {
-        var admAvailable = false
-        try {
-            Class.forName("com.amazon.device.messaging.ADM")
-            admAvailable = true
-        } catch (e: ClassNotFoundException) {
-            android.util.Log.i(DEFAULT_LOGTAG, "ADM is not available on this device.")
-        }
-
+        var admAvailable = ADM(applicationContext).isSupported
         // Only install push feature if ADM is available on this device
         if (admAvailable) {
+            Log.log(tag = DEFAULT_LOGTAG, message = "ADM not available")
             // This installs the [pushFeature] as a singleton
```

Comment:
nit: this comment is redundant to the implementation, which this calling code shouldn't be aware of anyway.